jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -168,6 +168,26 @@ public Response getSubscription(@PathParam("subscriptionId") final UUID subscrip
         return Response.status(Status.OK).entity(json).build();
     }
 
+    @TimedResource
+    @GET
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve a subscription by external key", response = SubscriptionJson.class)
+    @ApiResponses(value = {@ApiResponse(code = 404, message = "Subscription not found")})
+    public Response getSubscriptionByKey(@ApiParam(required = true) @QueryParam(QUERY_EXTERNAL_KEY) final String externalKey,
+                                         // The required part is a bit forceful for the user though... but removing would require not using our CatalogInternalApi down the line
+                                         // which was made specifically to ensure that catalog plugin get what they need.
+                                         @ApiParam(required = true) @QueryParam(QUERY_ACCOUNT_ID) final UUID accountId,
```

Comment:
I don't think we should require the `accountId`: the point of this API is to retrieve the subscription with only the key (the associated account might not be known).

Take a look at how we do it for `getActiveSubscriptionBundleForExternalKey`: there's a dao call to first get the mapping between the key and the id. From the id, we can then build the right context (`createInternalTenantContext(final UUID objectId, final ObjectType objectType, final TenantContext context)`) and plug into the methods to fully populate the object.

Reply:
ack, see https://github.com/killbill/killbill/commit/ad582da82dde399c0c60aa3c40a346b6a0551532